apollo-router/src/plugins/telemetry/mod.rs

```diff
@@ -256,116 +267,166 @@ impl Plugin for Telemetry {
-                            if let Some(header_name) = &expose_trace_id.header_name {
-                                let mut headers: HashMap<String, Vec<String>> = HashMap::new();
-                                if let Some(value) = response.response.headers().get(header_name) {
-                                    headers.insert(header_name.to_string(), vec![value.to_str().unwrap_or_default().to_string()]);
-                                    let response_headers = serde_json::to_string(&headers).unwrap_or_default();
-                                    span.record("apollo_private.http.response_headers",&response_headers);
+                        let expose_trace_id = &config.tracing.response_trace_id;
+                        if let Ok(response) = &response {
+                            span.set_dyn_attributes(
+                                config.spans.router.attributes.on_response(response),
+                            );
+                            if expose_trace_id.enabled {
+                                if let Some(header_name) = &expose_trace_id.header_name {
+                                    let mut headers: HashMap<String, Vec<String>> =
+                                        HashMap::with_capacity(1);
+                                    if let Some(value) =
+                                        response.response.headers().get(header_name)
+                                    {
+                                        headers.insert(
+                                            header_name.to_string(),
+                                            vec![value.to_str().unwrap_or_default().to_string()],
+                                        );
+                                        let response_headers =
+                                            serde_json::to_string(&headers).unwrap_or_default();
+                                        span.record(
+                                            "apollo_private.http.response_headers",
+                                            &response_headers,
+                                        );
+                                    }
                                 }
                             }
-                        }
 
-                        if response.response.status() >= StatusCode::BAD_REQUEST {
-                            span.record("otel.status_code", "Error");
-                        } else {
-                            span.record("otel.status_code", "Ok");
+                            if response.response.status() >= StatusCode::BAD_REQUEST {
+                                span.record(OTEL_STATUS_CODE, "Error");
+                            } else {
+                                span.record(OTEL_STATUS_CODE, "Ok");
+                            }
+                        } else if let Err(err) = &response {
+                            span.set_dyn_attributes(config.spans.router.attributes.on_error(err));
```

Comment:
Should we also add `span.record(OTEL_STATUS_CODE, "Error");` here ?

Reply:
Yes